Symfony/CS/Tests/Console/TextDiffTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tests\Console;
+
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\CS\Console\Command\FixCommand;
+
+/**
+ * @author SpacePossum
+ *
+ * @internal
+ */
+final class TextDiffTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @param bool   $decorated
+     * @param string $format
+     *
+     * @dataProvider provideDiffReporting
+     */
+    public function testDiffReportingDecorated($decorated, $format)
+    {
+        $command = new FixCommand();
+        $commandTester = new CommandTester($command);
+        $commandTester->execute(
+            array(
+                'path' => __DIR__.'/TextDiffTestInput.test',
+                '--fixers' => 'concat_without_spaces',
+                '--dry-run' => true,
+                '--diff' => true,
+            ),
+            array(
+                'decorated' => $decorated,
+                'verbosity' => OutputInterface::VERBOSITY_NORMAL,
+            )
+        );
+
+        if ($decorated !== $commandTester->getOutput()->isDecorated()) {
+            $this->markTestSkipped(sprintf('Output should %sbe decorated.', $decorated ? '' : 'not'));
+        }
+
+        $this->assertStringMatchesFormat($format, $commandTester->getDisplay(true));
+    }
+
+    public function provideDiffReporting()
+    {
+        $cases = array(
+            array(
+                false,
+                "%a-        \$output->writeln('' . 'a' . '');
++        \$output->writeln(''.'a'.'');%a",
+            ),
+        );
+
+        if (!getenv('APPVEYOR')) {
+            $cases[] = array(
+                true,
+                "%aTextDiffTestInput.test%a\033[%s-\033[%sm        \$output->writeln('<error>' . 'a' . '</error>');
```

Comment:
maybe prepare this by OutputInterface instead of hardcoding the colors and it should be fine for both system?